_posts/george_mihaiescu/2018-03-13-Quick-Kubernetes-installation.md

```diff
@@ -0,0 +1,333 @@
+---
+layout: post
+title:  "Scripted Kubernetes installation with dasboard and monitoring."
+breadcrumb: true
+author: george_mihaiescu
+date: 2018-03-21
+categories: george_mihaiescu
+tags:
+    - OpenStack
+    - Collaboratory
+    - Kubernetes
+teaser:
+    info: A quick how-to on deploying Kubernetes on top of Collaboratory VMs with Kubernetes dashboard and Graphana monitoring enabled.
+    image: george_mihaiescu/Kube/kubernetes-dashboard.png
+header:
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+tl;dr
+
+Step 1: Start an instance that will serve as your Kubernetes master and dump a config script.  
+Step 2: Start one or more instances that will serve as your Kubernetes worker nodes and dump a config script.  
+Step 3: Start using Kubernetes.  
+
+
+## Introduction
+
+In the previous blog [How-to-run-Kubernetes-in-Collaboratory](http://softeng.oicr.on.ca/george_mihaiescu/2017/09/28/How-to-run-Kubernetes-in-Collaboratory/) I described how Kubernetes can be deployed using Ansible, but in this blog I will show how you can use cloud-init to accomplish the same goal while also adding the Kubernetes dashboard and monitoring.  
+This blog post will not cover making Kubernetes highly available, as it's a more advanced topic reserved for a future blog post.
+
+Kubernetes has a server-client architecture, with two main types of servers:
+
+- A main node running the Kubernetes API, Etcd, scheduler, Kube-DNS and other management services.
+- One or more worker nodes running Kubelet and Kube-proxy where the Docker containers deployed by the user will be scheduled to run.
+
+First, go to the "Network, Security Group" tab and create a new security group called "All_TCP_access".
+Add a rule allowing access to all TCP ports from your source IP address (you can find your address by visiting [https://www.whatismyip.com/](https://www.whatismyip.com/).
+
+<figure>
+    <img src="{{site.urlimg}}george_mihaiescu/Kubernetes/sec_rule.png" />
```

Comment:
These are a lot of arrows :).
If you crop your picture, you would be able to increase its size to improve readability. In its current resolution your are pointing to a field but it's difficult to see the field name.